src/lib/nvme_pt_ioctl.c

```diff
@@ -2184,29 +2182,24 @@ int opal_ds_add_anybody_get(struct sed_device *dev, const char *key, uint8_t key
 	return ret;
 }
 
-int opal_list_lr_pt(struct sed_device *dev, const char *password, uint8_t key_len)
+int opal_list_lr_pt(struct sed_device *dev, const struct sed_key *key,
+		    struct sed_opal_lockingrange_arr *lrs)
 {
-	struct sed_key disk_key;
 	struct opal_device *opal_dev;
 	int ret = 0;
 
-	if (password == NULL) {
+	if (key == NULL) {
 		SEDCLI_DEBUG_MSG("Must provide the password.\n");
 		return -EINVAL;
 	}
 
 	opal_dev = dev->priv;
 
-	ret = sed_key_init(&disk_key, password, key_len);
-	if (ret) {
-		return ret;
-	}
-
-	ret = opal_start_admin1_lsp_session(dev->fd, opal_dev, &disk_key);
+	ret = opal_start_admin1_lsp_session(dev->fd, opal_dev, key);
 	if (ret)
 		goto end_sessn;
 
-	ret = list_lr(dev->fd, opal_dev);
+	ret = list_lr(dev->fd, opal_dev, lrs);
```

Comment:
I think it would be better to check the lrs pointer for NULL before it is being passed.